### PR TITLE
add : statoshi node statistics service

### PR DESCRIPTION
**Pull request by RandyMcMillan**
This service doesn't link to the signet services.
Runs as separate service.
Screen Shot 2021-09-11 at 2 47 46 PM

clear warning 

Screen Shot 2021-09-11 at 2 46 43 PM

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -7,26 +7,29 @@ export THIS_FILE
...
 export TIME
 
+#REF: https://wiki.debian.org/Multiarch/Tuples
 ARCH                                    :=$(shell uname -m)
 export ARCH
 ifeq ($(ARCH),x86_64)
-ARCH                                    :=x86_64-linux-gnu
+TRIPLET                                 :=x86_64-linux-gnu
+export TRIPLET
 export ARCH
```
Comment (RandyMcMillan) [inline GNUmakefile:16]: We needed to use conventional triplets to distinguish from ARCH

File: bitcoin-signet/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 FROM debian:buster-slim as builder
 
 ARG BITCOIN_VERSION=0.21.1
-ARG ARCH=${ARCH}
+ARG TRIPLET=${TRIPLET}
```
Comment (RandyMcMillan) [inline bitcoin-signet/Dockerfile:5]: We needed to use conventional triplets to distinguish from ARCH

File: bitcoin-signet/Dockerfile [hunk 3]
```diff
@@ -13,8 +13,8 @@ RUN  apt-get update && \
 WORKDIR /tmp
 
 # install bitcoin binaries
-RUN BITCOIN_URL="https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/bitcoin-${BITCOIN_VERSION}-${ARCH}.tar.gz" && \
-    BITCOIN_FILE="bitcoin-${BITCOIN_VERSION}-${ARCH}.tar.gz" && \
+RUN BITCOIN_URL="https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/bitcoin-${BITCOIN_VERSION}-${TRIPLET}.tar.gz" && \
+    BITCOIN_FILE="bitcoin-${BITCOIN_VERSION}-${TRIPLET}.tar.gz" && \
     BITCOIN_SHA_URL="https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS.asc" && \
```
Comment (RandyMcMillan) [inline bitcoin-signet/Dockerfile:18]: We needed to use conventional triplets to distinguish from ARCH

**Comment by RandyMcMillan** [comment]
@asherp - please test - the change to TRIPLET will probably break stuff - but is better long term.

**Comment by RandyMcMillan** [comment]
NOTE: aarch64 support not implemented yet. This PR anticipates aarch64 support in the future.

File: GNUmakefile [hunk 4]
```diff
@@ -7,26 +7,29 @@ export THIS_FILE
...
-#export HOST_UID
+ifeq ($(user),)
+HOST_USER								:= root
+HOST_UID								:= $(strip $(if $(uid),$(uid),0))
+else
+HOST_USER								:=  $(strip $(if $(USER),$(USER),nodummy))
+HOST_UID								:=  $(strip $(if $(shell id -u),$(shell id -u),4000))
+endif
+export HOST_USER
+export HOST_UID
```
Comment (RandyMcMillan) [inline GNUmakefile:32]: This can be used for other docker images.

**Comment by RandyMcMillan** [comment]
statoshi screenshot
Screen Shot 2021-09-11 at 3 18 51 PM

statoshi_datadir
Screen Shot 2021-09-11 at 3 19 41 PM

**Comment by rsafier** [comment]
pending platform support and also templating setup so we don't feature creep and bloat this too much

**Pull request closed without merging**